### PR TITLE
Arm7hf arch

### PR DESCRIPTION
**Pull request by Invertisment**
I've wanted to run selenium on arm7hf architecture and I wanted to use your code.
I hacked my way around and if you think that this could be used in the future we can talk about the changes. 
I don't think that this code is nowhere near the production quality, because I didn't add any configuration options. I only added a possibility to disregard OS and use arm7hf architecture. 
Do you think this code is useful? 

I used Firefox webdriver and it has the required arm binaries.

### CODE REVIEW

File: src/main/java/io/github/bonigarcia/wdm/Architecture.java [hunk 1]
```diff
@@ -23,7 +23,7 @@
  * @since 1.0.0
  */
 public enum Architecture {
-    DEFAULT, X32, X64;
+    DEFAULT, X32, X64, X7HF;
```
Comment (bonigarcia) [inline src/main/java/io/github/bonigarcia/wdm/Architecture.java]: The new architecture value should be ARMv7h instead.

File: src/main/java/io/github/bonigarcia/wdm/OperatingSystem.java [hunk 2]
```diff
@@ -23,5 +23,5 @@
  * @since 1.0.0
  */
 public enum OperatingSystem {
-    WIN, LINUX, MAC
+    WIN, LINUX, MAC, ARM;
```
Comment (bonigarcia) [inline src/main/java/io/github/bonigarcia/wdm/OperatingSystem.java]: ARM is not an operating system.

File: src/main/java/io/github/bonigarcia/wdm/Config.java [hunk 3]
```diff
@@ -165,19 +166,27 @@ public void reset() {
...
+    private boolean isArm() {
+        return System.getProperty("os.arch").equals("arm");
+    }
+
     private String defaultOsName() {
-        String osName = System.getProperty("os.name").toLowerCase();
         if (IS_OS_WINDOWS) {
-            osName = WIN.name();
+            return WIN.name();
+        } else if (isArm()) {
```
Comment (bonigarcia) [inline src/main/java/io/github/bonigarcia/wdm/Config.java]: ARM is not an operating system.

File: src/main/java/io/github/bonigarcia/wdm/UrlFilter.java [hunk 4]
```diff
@@ -45,6 +45,7 @@
     final Logger log = getLogger(lookup().lookupClass());
 
     public List<URL> filterByOs(List<URL> list, String osName) {
+        System.out.println("Filtering drivers by OS: " + osName);
```
Comment (bonigarcia) [inline src/main/java/io/github/bonigarcia/wdm/UrlFilter.java]: Do not use System.out.println for logging. Use log instead (in fact it was already used, at trace level).

**Comment by bonigarcia** [comment]
According to this [page](https://github.com/mozilla/geckodriver/issues/1231), the architecture is ARMv7h and the operative system is Linux. Therefore, part of your PR needs to be reviewed. See comments inline.

File: src/main/java/io/github/bonigarcia/wdm/Architecture.java (hunk 1 above)
Comment (Invertisment) [reply src/main/java/io/github/bonigarcia/wdm/Architecture.java]: These hacks were made because firefox doesn't name the package correctly.

**Comment by Invertisment** [comment]
This was a hack that would make it possible to use firefox webdriver on arm, because they use different naming schema.
I'm not sure if this is mergeable at all, because all types would need to be refactored.

**Pull request closed without merging**